public/mobile-app/src/routes/address/+page.svelte

```diff
@@ -1,17 +1,38 @@
 <script lang="ts">
+  import { onMount } from 'svelte'
+  import { goto } from '$app/navigation'
   import { Address } from '$lib/address'
   import { type AddressFromBAN, callBAN } from '$lib/addressesFromBAN'
   import { userStore } from '$lib/state/User.svelte'
 
+  let addressFromUserStore: Address | null = $state(null)
   let timer: any
   let inputValue: string = $state('')
   let filteredAddresses: Address[] = $state([])
   let disabledButton: boolean = $state(true)
-  let addressHasError: boolean = $state(false)
+  let addressApiHasError: boolean = $state(false)
+  let addressInputHasError: boolean = $state(false)
   let selectedAddress: Address = $state<Address>(new Address())
   let hasSubmittedAddress: boolean = $state(false)
   let submittedAddress: Address = $state<Address>(new Address())
 
+  onMount(async () => {
+    if (!userStore.connected) {
+      goto('/')
+      return
+    } else {
```

Comment:
Nit: no need for the `else` as there is an early return line 22, this avoid having the extra indentation.